modules/docs/markdown/06-guides/testing.md

```diff
@@ -0,0 +1,41 @@
+
+Note that if you are going to test Smithy4s interpreter behaviors, you should _not_ use Smithy4s to test them. We have occasionally seen the pattern where someone uses a Smithy4s client to test a Smithy4s server. This type of testing has very little benefit since much of the same logic is used for each of these interpreters. This means that the interpreters you are testing could both have the same bug and your tests would not catch it.
+
+On the other hand, we do recommend that you test the implementation of your service. For example, if Smithy4s generates a service interface called `HelloWorldService`, you will implement that interface and place your business logic within it. We recommend that you test all of your business logic since this is outside the scope of what Smithy4s controls.
+
+## Schema and Contract Testing
+
+Contract testing is a form of testing that checks that the contracts which services and clients agree upon are stable over time. Note that schema-based tests and contract-based test are not the same thing. Schema tests are concerned with checking that the data sent and received conforms to a specified schema or schemas. An example where contract testing differs is, for example, a client may expect a certain error response to be returned whenever certain conditions are met in the request. A schema can capture what errors may possibly be sent, but it will not capture which error should be triggered by a specific request.
+
+For your specific use case, you may have a need to employ schema-based and/or contract-based testing. Schema based testing would be one way of checking that schema changes over time do not impact existing clients. Contract-based testing could accomplish this same thing and more. However, contract based testing is usually more difficult to implement than schema-based testing so you may find it not to be worth it for your use case. In either case, remember to make sure you know _what_ it is you are testing and what you are not.
+
+## TL;DR
+
+Ultimately our recommendation is that you unit test all of your business logic as well as any custom implementations you have created on top of Smithy4s. You should be able to leave the rest of the unit testing up to us. However, your requirements, concerns, and experience may vary and we encourage you to do whatever will help you write the best software.
+
+Depending on your use case, you may find value in Schema or Contract testing. These methods can help you make sure your API is stable for consumers over time.
```

Comment:
**suggestion:** can we recommend some resources on how to do schema and contract testing?

Reply:
Sure, do you have any in mind? I know pact is one, but I think that is a commercial thing (not positive) and I don't really love pushing people toward stuff that isn't OSS 🤷